### PR TITLE
bpo-29539: Implementation to return all mta status codes

### PR DESCRIPTION
**Pull request by ghost**
- Refactored senderrs
- sendmail returns all mta status codes for each recipient
- Updated docstrings
- Implemented unittests

https://bugs.python.org/issue29539

### CODE REVIEW

**Comment by remilapeyre** [comment]
Hi @sls89, the tests should be able to tun from the CI.

I think you could takee inspiration from `SMTPAUTHInitialResponseSimTests` to not rely on an external mail server.

You will also need to add a blurb (https://devguide.python.org/committing/#what-s-new-and-news-entries) to describe your changes.

You can use https://blurb-it.herokuapp.com/ to do so.

**Comment by sls89** [comment]
Hi @remilapeyre 

Thanks for your reply. I updated the unittests with a mock for smtp-sessions as well as a news entry. 
Is there anything left?

**Comment by remilapeyre** [comment]
Hi @sls89, thanks for making the changes. Some tests broke like `test_name_field_not_included_in_envelop_addresses`, can you have a look at them?

I think it's because you did not hide this behavior behind a flag like in David Ford's proposal (https://bugs.python.org/issue29539#msg287662). This is not backward compatible and I don't think the new feature can be accepted as is. Can you add new new parameter so this new feature is disabled by default and can be opted-in?

**Comment by sls89** [comment]
@remilapeyre 

You're right, sorry, I fixed that. All smtplib-tests should be ok now.

File: Misc/NEWS.d/next/Library/2019-03-10-14-23-58.bpo-29539.HB-mqJ.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+smtplib.SMTP.sendmail() now returns both error and success messages.
```
Comment (remilapeyre) [inline Misc/NEWS.d/next/Library/2019-03-10-14-23-58.bpo-29539.HB-mqJ.rst]: Add `Patch contributed by ...` here.

**Review by remilapeyre — CHANGES_REQUESTED** [review]
Thanks, the code and the tests look good now.

Last thing, can you update the documentation at https://github.com/python/cpython/blob/master/Doc/library/smtplib.rst ?

**Comment by sls89** [comment]
I made the changes, thanks for your guidance!

**Comment by sls89** [comment]
Hi Guys,

is there any progress?

**Comment by orennu** [comment]
hi,
any update on this feature? really looking forward to have it.

**Pull request closed without merging**